docs/core_apis.md

```diff
@@ -32,16 +32,22 @@ You can configure your Data Prepper core APIs through the `data-prepper-config.y
 
 Many of the Getting Started guides in this project disable SSL on the endpoint.
 
-```
+```yaml
 ssl: false
 ```
 
 To enable SSL on your Data Prepper endpoint, configure your `data-prepper-config.yaml`
 with the following:
 
-```
+```yaml
 ssl: true
-keyStoreFilePath: "/usr/share/data-prepper/keystore.jks"
+keyStoreFilePath: "/usr/share/data-prepper/keystore.p12"
 keyStorePassword: "secret"
 privateKeyPassword: "secret"
 ```
+
+If you are using a self-signed certificate, you must add the `-k` flag to curl requests.
```

Comment:
QUES: Just curious. Is there a way that providing a client cert might work for self-signed certificate? i.e.

```
curl --cert *.pem ...
```

Reply:
One option is to replace the `-k` flag with `--cacert certfile.crt` where `certfile.crt` is the server's certificate that is stored in the keyStore in this case. However this requires that all of the info suppllied to the csr matches up with the client request, such as the host being `localhost`. 

Like you mentioned, the `--cert` flag is the client certificate. It is essentially another form of authentication for the client to provide to the server. It seems we don't require client certificates on the Data Prepper side of things, so I don't think this bypasses the `-k` flag, as from my understanding the `-k` flag is just the client saying that they don't care about the validity of the server, and they want to connect no matter what.

Reply:
Providing a client certificate to cURL does make this request secure if the certificate is not the one which we have provided. So I think there is value in including `--cert` instead. But, it also appears that this current document doesn't yet reference making your own self-signed certificate.

Reply:
True. This is just the core_apis doc, and I figured I would provide the `-k` in order to save people some time. I should probably link to the configuration.md from here though since I am showing a `data-prepper-config.yaml` with SSL enabled.